client/src/actions/actions.js

```diff
@@ -15,31 +16,18 @@ export const updateLink = (newLink) => {
     };
 };
 
-export const addSong = (link) => (dispatch) => {
-    dispatch({ type: BEGIN_DOWNLOAD_SONG_INFO, link });
-    const videoId = getVideoIdFromLink(link);
-    const url = formatString(config.server.resources.getVideoInfo, [videoId]);
-    fetch(url, {
-        method: 'GET',
-        headers: {
-            Accept: 'application/json'
-        },
-    })
-    .then(response => {
-        if (response.ok) {
-            response.json().then(songInfo => {
-                dispatch(downloadSongInfoSuccess(songInfo));
-            })
-        }
-        else {
-            response.json().then(jsonBody => {
-                dispatch(downloadSongInfoError(jsonBody.message));
-            })
-        }
-    })
-    .catch((e) => {
-        dispatch(downloadSongInfoError(config.download_song_info_error_message))
-    });
+export const addSong = songId => async dispatch => {
+    dispatch({ type: BEGIN_DOWNLOAD_SONG_INFO, songId });
+    const url = config.server.url + formatString(config.server.resources.getVideoInfo, songId);
+    const callbackError = () => dispatch(downloadSongInfoError(config.messages.downloadSongInfoErrorMessage));
+    const statusActions = {
+        200: responseBody => dispatch(downloadSongInfoSuccess(responseBody)),
```

Comment:
Why not save the status code in configuration? Not everyone memorize the response codes

Reply:
I swear I thought about that.
do you know  the package `http-status-codes`?

Reply:
yes - you can use it (make sure it has a small size , if it doesn't then implement it yourself